File: core/src/main/java/net/siegerpg/siege/core/drops/materials/VINE.java
package net.siegerpg.siege.core.drops.materials;

import net.siegerpg.siege.core.drops.BlockDropTable;
import net.siegerpg.siege.core.drops.Reward;
import net.siegerpg.siege.core.items.implemented.misc.materials.drops.blocks.Vine;
import org.bukkit.Material;

public class VINE extends BlockDropTable {

	public VINE() {

		super(40, Material.VINE, 0, 0, 0, 0, new Reward[] {
				new Reward(new Vine()
						           .getUpdatedItem(false), 100.0),
				new Reward(new Vine()
						           .getUpdatedItem(false), 10.0),
				new Reward(new Vine()
						           .getUpdatedItem(false), 0.1)
		});
	}

}

File: core/src/main/kotlin/net/siegerpg/siege/core/items/implemented/misc/materials/drops/blocks/Vine.kt
package net.siegerpg.siege.core.items.implemented.misc.materials.drops.blocks

import net.siegerpg.siege.core.items.CustomItemUtils
import net.siegerpg.siege.core.items.enums.Rarity
import net.siegerpg.siege.core.items.types.misc.CustomMaterial
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class Vine() : CustomMaterial(
		name = "Vine",
		customModelData = 320004,
		description = listOf("Woven around trees", "and abandoned lands"),
		levelRequirement = 0,
		material = Material.FLINT,
		) {
	override fun getSellValue() : Int {
		return 2
	}
	constructor(quality : Int) : this() {
		this.quality = quality
		this.rarity = Rarity.getFromInt(quality)
		this.serialize()
	}

	constructor(item : ItemStack) : this() {
		this.item = item
		deserialize()
	}

}